NameLayerMain/src/vg/civcraft/mc/namelayer/group/Group.java

```diff
@@ -317,23 +318,38 @@ public PlayerType getCurrentRank(UUID uuid) {
 	 * @param type- The PlayerType to add. If a preexisting PlayerType is found, 
 	 * it will be overwritten.
 	 */
-	public void addMember(UUID uuid, PlayerType type) {
+	
+	public void addMember(UUID uuid, PlayerType type){
+		addMember(uuid,type,true);
+	}
+	
+	public void addMember(UUID uuid, PlayerType type, boolean savetodb) {
 		if (type == PlayerType.NOT_BLACKLISTED) {
 			return;
 		}
-		if (isMember(uuid, type)) {
-			db.removeMember(uuid, name);
+		if (savetodb == true) {
+			if (isMember(uuid, type)){
+				db.removeMemberAsync(uuid, name);
```

Comment:
In this specific case, since we're talking about promotion (right?) just add a new UPDATE query, so that it's atomic at the DB level, and there's no issue.